pkg/kubelet/stability/manager_linux.go

```diff
@@ -0,0 +1,287 @@
+	containerName := "/"
+	cadvisorOptions := v2.RequestOptions{IdType: v2.TypeName, Count: 1, Recursive: false}
+	containerInfoResults, err := m.cadvisorInterface.ContainerInfoV2(containerName, cadvisorOptions)
+	if err != nil {
+		glog.Errorf("stability manager: unexpected err: %v", err)
+		return
+	}
+	containerInfo, found := containerInfoResults[containerName]
+	if !found {
+		glog.Errorf("stability manager: container info %v not found", containerName)
+	}
+
+	// build eviction signals
+	// TODO: create a typedef
+	// TODO: add future signal support for cpu load, etc.
+	observations := map[componentconfig.EvictionSignal]resource.Quantity{}
+	if availableMemory, err := calculateMemoryAvailable(&containerInfo); err != nil {
+		glog.Errorf("stability manager: unable to calculate available memory: %v", err)
+	} else {
+		observations[componentconfig.EvictionMemoryAvailable] = availableMemory
+	}
+
+	// see if any eviction threshold has been met
+	evictionObservations := []EvictionObservation{}
+	for _, threshold := range m.evictionThresholds {
+		glog.Infof("stability manager: debug - signal %s, threshold: %s, op: %v", threshold.Signal, threshold.Value.String(), threshold.Operator)
+		observed, found := observations[threshold.Signal]
+		if !found {
+			glog.Warningf("stability manager: no observation found for signal %v", threshold.Signal)
+			continue
+		}
+		glog.Infof("stability manager: debug - signal %s, observed: %s", threshold.Signal, observed.String())
+
+		result := threshold.Value.Cmp(observed)
+		thresholdMet := false
+		if componentconfig.EvictionOpGreaterThan == threshold.Operator && result < 0 {
+			thresholdMet = true
+		}
+		if componentconfig.EvictionOpLessThan == threshold.Operator && result > 0 {
+			thresholdMet = true
+		}
+		glog.Infof("stability manager: debug - signal %s, thresholdMet: %v", threshold.Signal, thresholdMet)
+
+		if !thresholdMet {
+			continue
+		}
+
+		// TODO: add support for graceperiod, right now, everything is a hard eviction
+		evictionObservation := EvictionObservation{
+			EvictionThreshold: componentconfig.EvictionThreshold{
+				Signal:   threshold.Signal,
+				Value:    *(threshold.Value.Copy()),
+				Operator: threshold.Operator,
+			},
+			Observed: observed,
+		}
+		evictionObservations = append(evictionObservations, evictionObservation)
+	}
+
+	if len(evictionObservations) == 0 {
+		glog.Infof("stability manager: debug - no evictions are required")
+		return
+	}
+
+	// Find a candidate pod to evict from the strategy
+	// Kill the pod and wait until its dead
+	evictionOptions := &evictionOptionsImpl{
+		podLifecycleManager: podLifecycleManager,
+		resourceAccountant:  m.resourceAccountant,
+		observations:        evictionObservations,
+	}
+	pod, err := m.evictionStrategy.Select(evictionOptions)
+	if err != nil {
+		glog.Errorf("stability manager: unable to find pod to evict %v", err)
+		return
+	}
+	glog.Infof("stability manager: kill pod %s", pod.Name)
+	err = podLifecycleManager.KillPod(pod)
```

Comment:
recording a note here so when i get time to get back to this PR, but I will look to use `kubelet.HandlePodDeletions`